aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterClearTest.java

```diff
@@ -0,0 +1,111 @@
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.storage.sqlite;
+
+import android.content.Context;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.datastore.StrictMode;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.datastore.storage.SynchronousStorageAdapter;
+import com.amplifyframework.testmodels.commentsblog.AmplifyModelProvider;
+import com.amplifyframework.testmodels.commentsblog.BlogOwner;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+
+import io.reactivex.observers.TestObserver;
+
+import static com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.DATABASE_NAME;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class SQLiteStorageAdapterClearTest {
+    private SynchronousStorageAdapter adapter;
+    private Context context;
+    private TestObserver<StorageItemChange.Record> observer;
+
+    @BeforeClass
+    public static void enableStrictMode() {
+        StrictMode.enable();
+    }
+
+    @Before
+    public void setup() {
+        TestStorageAdapter.cleanup();
+        context = (ApplicationProvider.getApplicationContext());
+        adapter = TestStorageAdapter.create(AmplifyModelProvider.getInstance());
+        observer = TestObserver.create();
+        adapter.observe().subscribe(observer);
+    }
+
+    @After
+    public void teardown() {
+        TestStorageAdapter.cleanup(adapter);
+    }
+
+    /**
+     * Happy path test: saves a record to the database
+     * then calls clear and verifies that the database
+     * file has been deleted.
+     * @throws DataStoreException bubbles up exceptions thrown from the adapter
+     */
+    @Test
+    public void clearDeletesDatabaseFromDisk() throws DataStoreException {
+        final BlogOwner dummy = BlogOwner.builder()
+            .name("Dummy Blogger")
+            .build();
+
+        adapter.save(dummy);
+        List<BlogOwner> query = adapter.query(BlogOwner.class);
+        assertEquals(1, query.size());
+        //Check that exactly one event was received by the observer.
+        observer.assertValueCount(1);
+
+        adapter.clear();
+        assertFalse(context.getDatabasePath(DATABASE_NAME).exists());
+        //check that it's not subscribed anymore.
+        observer.assertNotSubscribed();
+        //Assert the observer was terminated during the file removal.
+        observer.assertTerminated();
```

Comment:
is there a way to assert that the success handler gets called?

Reply:
We created a [wrapper class](https://github.com/aws-amplify/amplify-android/blob/782eb601676c1e94b85718c4a66028784e90a802/aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/SynchronousStorageAdapter.java#L296) for that we use for testing that basically blocks until either the success handler gets called or bubbles up an exception. So if the call does not throw, we know the success handler was called.